ambry-store/src/main/java/com/github/ambry/store/IndexSegment.java

```diff
@@ -258,17 +269,24 @@ int size() {
   }
 
   /**
-   * @return The persisted entry size of this segment
+   * @return The entry size of this segment
+   * It's for testing purpose only.
    */
   int getPersistedEntrySize() {
-    return persistedEntrySize;
+    rwLock.readLock().lock();
+    int sz = sealed.get() ? sealedIndex.getPersistedEntrySize() : maxEntrySize;
```

Comment:
so the entry size can change if the index segment is not sealed ? and once its sealed all entry sizes are same ? how is that possible ?

Reply:
Each entry can have different size. That's why we use maxEntrySize to monitor the max entry size.
When we persist, we use the maxEntrySize to do the padding. If the entry is smaller than that size, we pad the zero to it.

Reply:
oh, ok that explains why all entries are of the same size.